File: TransCoder_tokenized_test_set_functions/java/K_TH_LARGEST_SUM_CONTIGUOUS_SUBARRAY.java
static int kthLargestSum ( int arr [ ], int n, int k ) {
  int sum [ ] = new int [ n + 1 ];
  sum [ 0 ] = 0;
  sum [ 1 ] = arr [ 0 ];
  for ( int i = 2;
  i <= n;
  i ++ ) sum [ i ] = sum [ i - 1 ] + arr [ i - 1 ];
  PriorityQueue < Integer > Q = new PriorityQueue < Integer > ( );
  for ( int i = 1;
  i <= n;
  i ++ ) {
    for ( int j = i;
    j <= n;
    j ++ ) {
      int x = sum [ j ] - sum [ i - 1 ];
      if ( Q . size ( ) < k ) Q . add ( x );
      else {
        if ( Q . peek ( ) < x ) {
          Q . poll ( );
          Q . add ( x );
        }
      }
    }
  }
  return Q . poll ( );
}

File: results/py/K_TH_LARGEST_SUM_CONTIGUOUS_SUBARRAY.cpp
int f_filled ( int * arr , int n , int k ) {
  int sum [ n + 1 ] ;
  sum [ 0 ] = 0 ;
  sum [ 1 ] = arr [ 0 ] ;
  for ( int i = 2 ;
  i <= n ;
  i ++ ) sum [ i - 1 ] = sum [ i - 1 ] + arr [ i - 1 ] ;
  int * Q = new int [ n + 1 ] ;
  int * heapQ = new int [ n + 1 ] ;
  for ( int i = 1 ;
  i <= n ;
  i ++ ) {
    for ( int j = i ;
    j <= n ;
    j ++ ) {
      int x = sum [ j ] - sum [ i - 1 ] ;
      if ( Q [ 0 ] < x ) heapQ [ 0 ] = x ;
      else {
        if ( heapQ [ 0 ] < x ) {
          heapQ [ 0 ] = x ;
          heapQ [ 0 ] = x ;
        }
      }
    }
  }
  delete [ ] heapQ ;
  return Q [ 0 ] ;
}
